### PR TITLE
PEO-342 Implement LolDba runner

### PR DESCRIPTION
**Pull request by rostikkkk2**
[PEO-342](https://garage.atlassian.net/browse/PEO-342)

### Description
Need to implement lol_dba runner.
By default each level of the issue equal low

### Before submitting the merge request make sure the following are checked

- [x] Followed the style guidelines of this project
- [x] Performed a self-review of own code
- [x] Wrote the tests that prove that fix is effective/that feature works

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -10,3 +10,9 @@ Metrics/LineLength:
 
 Style/FrozenStringLiteralComment:
   Enabled: false
+
+RSpec/AnyInstance:
+  Enabled: false
```
Comment (chubchenko) [inline .rubocop.yml]: These cops should be enabled

File: lib/inquisition.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require 'pry'
```
Comment (chubchenko) [inline lib/inquisition.rb]: Same

File: spec/fixtures/lol_dba/app/models/task.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Task < ActiveRecord::Base
```
Comment (chubchenko) [inline spec/fixtures/lol_dba/app/models/task.rb]: We don't need this file

File: spec/fixtures/lol_dba/schema.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+ActiveRecord::Schema.define do
```
Comment (chubchenko) [inline spec/fixtures/lol_dba/schema.rb]: Same

File: lib/inquisition/lol_dba/runner.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
+require 'lol_dba'
+
+module Inquisition
+  module LolDba
+    class Runner < ::Inquisition::Runner
+      LEVEL_LOW = :low
```
Comment (chubchenko) [inline lib/inquisition/lol_dba/runner.rb]: Why do we need this constant here? Not sure that it related to the runner

File: lib/inquisition/lol_dba/runner.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+      def create_issue_with_every_index(table, index_arr)
+        index_arr.map { |index| create_issue(table, index) }
+      end
+
+      def create_issue(table, index)
+        Inquisition::Issue.new(
+          level: LEVEL_LOW,
+          line: "migrate/#{table}",
+          runner: RUNNER_NAME,
```
Comment (chubchenko) [inline lib/inquisition/lol_dba/runner.rb]: `self` instead of `RUNNER_NAME`

File: lib/inquisition/lol_dba/runner.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      end
+
+      def create_issue_with_every_index(table, index_arr)
+        index_arr.map { |index| create_issue(table, index) }
+      end
+
+      def create_issue(table, index)
+        Inquisition::Issue.new(
+          level: LEVEL_LOW,
+          line: "migrate/#{table}",
```
Comment (chubchenko) [inline lib/inquisition/lol_dba/runner.rb]: line should be nil

File: lib/inquisition/lol_dba/runner.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+          runner: RUNNER_NAME,
+          file: "Table #{table}",
```
Comment (chubchenko) [inline lib/inquisition/lol_dba/runner.rb]: file should be nil

File: lib/inquisition/lol_dba/runner.rb [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+          message: "You have not index in column #{index}"
```
Comment (chubchenko) [inline lib/inquisition/lol_dba/runner.rb]: Message should contain information about table and column

File: lib/inquisition/lol_dba/runner.rb [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        )
+      end
+
+      def load_environment
+        require "#{Dir.pwd}/config/environment"
```
Comment (chubchenko) [inline lib/inquisition/lol_dba/runner.rb:35]: Why do we need to require env?

File: spec/inquisition/lol_dba/runner_spec.rb [hunk 11]
```diff
@@ -0,0 +1,32 @@
+RSpec.describe Inquisition::LolDba::Runner do
+  describe '#call' do
+    let(:instance) { described_class.new }
+
+    context 'when call with errors and without' do
```
Comment (chubchenko) [inline spec/inquisition/lol_dba/runner_spec.rb]: maybe we need to split in into 2 contexts, what do you think?

File: spec/inquisition/lol_dba/runner_spec.rb [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+      before do
+        allow_any_instance_of(described_class).to receive(:load_environment).and_return(true)
+      end
+
+      it 'without errors' do
```
Comment (chubchenko) [inline spec/inquisition/lol_dba/runner_spec.rb]: bad description

File: spec/inquisition/lol_dba/runner_spec.rb [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        allow(LolDba::IndexFinder).to receive(:check_for_indexes).and_return({})
+        expect(described_class.call).to eq([])
```
Comment (chubchenko) [inline spec/inquisition/lol_dba/runner_spec.rb]: https://relishapp.com/rspec/rspec-expectations/v/2-2/docs/matchers/predicate-matchers

File: spec/inquisition/lol_dba/runner_spec.rb [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+      end
+
+      it 'with errors class name' do
```
Comment (chubchenko) [inline spec/inquisition/lol_dba/runner_spec.rb]: 3 specs can be combined into 1 with match_array

**Review by chubchenko — CHANGES_REQUESTED** [review]

File: spec/spec_helper.rb [hunk 15]
```diff
@@ -1,6 +1,8 @@
 require 'simplecov'
 require 'inquisition'
 
+ENV['RAILS_ENV'] ||= 'test'
```
Comment (chubchenko) [inline spec/spec_helper.rb:4]: Why do we need it?

File: spec/inquisition/lol_dba/runner_spec.rb [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+          file: nil, level: :low, line: nil,
+          message: 'You have not index in table `tasks`, column `project_id`',
+          runner: be_kind_of(described_class)
+        ).and_call_original
+        expect(described_class.call.count).to eq(1)
+      end
+    end
+
+    context 'when call runner and it return not one error' do
+      it 'without errors' do
```
Comment (chubchenko) [inline spec/inquisition/lol_dba/runner_spec.rb:24]: ?

**Review by chubchenko — CHANGES_REQUESTED** [review]

**Comment by chubchenko** [comment]
@rostikkkk2 your branch has conflicts, can you resolve them please, thanks

**Review by Svatok — APPROVED** [review]

**Review by polykoi — APPROVED** [review]

**Pull request merged**